packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -169,10 +185,30 @@ function s(
     }
     if (
         !isUndefined(slotset) &&
-        !isUndefined(slotset[slotName]) &&
-        slotset[slotName].length !== 0
+        !isUndefined(slotset.slotAssignments) &&
+        !isUndefined(slotset.slotAssignments[slotName]) &&
+        slotset.slotAssignments[slotName].length !== 0
     ) {
-        children = slotset[slotName];
+        children = slotset.slotAssignments[slotName].reduce((acc, vnode) => {
+            // If the passed slot content is factory, evaluate it and use the produced vnodes
+            if (vnode && isVScopedSlotFragment(vnode)) {
+                const vmBeingRenderedInception = getVMBeingRendered();
+                let children: VNodes = [];
+                if (!isUndefined(slotset.owner)) {
+                    // Evaluate in the scope of the slot content's owner
+                    setVMBeingRendered(slotset.owner);
+                }
```

Comment:
This branch should always evaluate, in other words there shouldn't be a case where `owner` is undefined. See the comment below on the `SlotSet` interface.

Reply:
I could remove the `isUndefined` check and always assume slotset.owner is present. `setVMBeingRendered(slotset.owner!);`

Reply:
> I could remove the `isUndefined` check and always assume slotset.owner is present.

I think this is preferable since it better models the actual runtime behavior. It's impossible in active to have slotted content without an owner.